### PR TITLE
[16.0] Add auth_oidc_environment

### PR DESCRIPTION
**Pull request by adrienpeiffer**
From : https://github.com/OCA/server-env/pull/143

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
@AnizR @sbidoul @simahawk

**Review by AnizR — APPROVED** [review]
Great work! 
Code LGTM and already tested on the field (from https://github.com/OCA/server-env/pull/143)

File: auth_oidc_environment/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+To configure this module, you need to:
+
+Create a module server_environment_file with a cfg file or set the environment variable SERVER_ENV_CONFIG with the following section:
+
+[auth_oauth_provider.<name>]
```
Comment (simahawk) [inline auth_oidc_environment/readme/CONFIGURE.rst:5]: I always recommend to use tech_name instead which is more reliable

**Review by simahawk — APPROVED** [review]
LG

File: auth_oidc_environment/readme/CONFIGURE.rst (hunk 1 above)
Comment (sbidoul) [reply auth_oidc_environment/readme/CONFIGURE.rst:5]: Agreed. There is no such field on oauth providers, though.

Comment (simahawk) [reply auth_oidc_environment/readme/CONFIGURE.rst:5]: that's why we have the techname.mixin (which I wonder if we shouldn't make the default :thinking: )

Comment (sbidoul) [reply auth_oidc_environment/readme/CONFIGURE.rst:5]: Ah, interesting. Thanks for the tip!

**Pull request merged**